### PR TITLE
[12.0] [MIG] currency monthly rate

### PR DESCRIPTION
**Pull request by i-vyshnevska**
migrate to v12

### CODE REVIEW

**Review by Tonow-c2c — APPROVED** [review]
LGTM

**Review by sebalix — APPROVED** [review]

File: currency_monthly_rate/models/res_currency.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                                 FROM res_currency_rate_monthly r
+                                 WHERE r.currency_id = c.id
+                                 AND r.name <= %s
+                                 AND (r.company_id IS NULL
+                                      OR r.company_id = %s)
+                                 ORDER BY r.company_id, r.name DESC
+                                 LIMIT 1) AS rate
+                   FROM res_currency c
+                   WHERE c.id IN %s"""
+        self._cr.execute(query, (date, company_id, tuple(self.ids)))
```
Comment (simahawk) [inline currency_monthly_rate/models/res_currency.py]: would be better to use `self.env.cr`

File: currency_monthly_rate/models/res_currency.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        currency_rates = dict(self._cr.fetchall())
```
Comment (simahawk) [inline currency_monthly_rate/models/res_currency.py]: `self.env.cr.dictfetchall`?

File: currency_monthly_rate/models/res_currency.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        for rate in self:
+            rate.name = fields.Date.from_string('%s-%s-01' % (
+                rate.year, rate.month))
+
+    # The two constraints have the same message because, the first is a
+    # redefinition of the constraint inherited from res.currency.rate and
+    # the second makes it stronger
+    _sql_constraints = [
+        ('unique_name_per_day', 'unique (name,currency_id,company_id)',
+         _('Only one currency monthly rate per month allowed!')),
```
Comment (simahawk) [inline currency_monthly_rate/models/res_currency.py]: no need to translate here, it happens automatically

File: currency_monthly_rate/views/res_currency.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    </record>
+
+    <record id="view_currency_rate_monthly_form" model="ir.ui.view">
+        <field name="name">res.currency.rate.monthly.form</field>
+        <field name="model">res.currency.rate.monthly</field>
+        <field name="arch" type="xml">
+            <form string="Currency Monthly Rate">
+                <sheet>
+                    <group>
+                        <group>
```
Comment (simahawk) [inline currency_monthly_rate/views/res_currency.xml]: name the groups

**Review by simahawk — CHANGES_REQUESTED** [review]

**Pull request merged**